nifty-core/src/main/java/com/facebook/nifty/core/ConnectionContext.java

```diff
@@ -29,6 +29,13 @@
     public SocketAddress getRemoteAddress();
 
     /**
+     * Checks whether the client has disconnected.
+     *
+     * @return {@code true} if the client has disconnected
+     */
+    public boolean hasClientDisconnected();
```

Comment:
nit: I'd prefer this to have the opposite meaning, something like `isClientConnected` to be in line with what AFAIR we have in c++. Not a major concern though